gradoop-data-integration/src/main/java/org/gradoop/dataintegration/transformation/ExtractProperty.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.dataintegration.transformation;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.util.Preconditions;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.common.model.impl.pojo.Edge;
+import org.gradoop.common.model.impl.pojo.Vertex;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.dataintegration.transformation.config.EdgeDirection;
+import org.gradoop.dataintegration.transformation.functions.CreateNewEdges;
+import org.gradoop.dataintegration.transformation.functions.CreateNewVertex;
+import org.gradoop.dataintegration.transformation.functions.CreateNewVertexWithEqualityCondense;
+import org.gradoop.dataintegration.transformation.functions.ExtractPropertyWithOriginId;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.model.api.operators.UnaryGraphToGraphOperator;
+import org.gradoop.flink.model.impl.functions.epgm.ByLabel;
+import org.gradoop.flink.model.impl.functions.tuple.Value0Of2;
+
+import java.util.List;
+
+/**
+ * This operator is executed on the vertex dataset and extracts a property to a newly created vertex
+ * which can be connected to the vertex it was created from. If the property is a list every
+ * element of the list is a new vertex.
+ */
+public class ExtractProperty implements UnaryGraphToGraphOperator {
+
+  /** The vertices the extraction is executed for. */
```

Comment:
I will change this. However, IMHO this rule is contra productive because it reduces the readability of the code. I see your point for methods and so on. But not for member variables.